shopinvader_api_cart/schemas/cart.py

```diff
@@ -14,3 +14,45 @@ class CartTransaction(StrictExtendableBaseModel):
 
 class CartSyncInput(StrictExtendableBaseModel):
     transactions: List[CartTransaction]
+
+
+class ShippingUpdateInfo(StrictExtendableBaseModel):
+    address_id: int
+
+
+class InvoicingUpdateInfo(StrictExtendableBaseModel):
+    address_id: int
+
+
+class CartUpdateInput(StrictExtendableBaseModel):
+    client_order_ref: str | None = None
+    shipping: ShippingUpdateInfo | None = None
+    invoicing: InvoicingUpdateInfo | None = None
+    note: str | None = None
+
+    def _convert_shipping_values(self, values):
+        if "address_id" in values:
+            return {"partner_shipping_id": values["address_id"]}
+        else:
+            return {}
+
+    def _convert_invoicing_values(self, values):
+        if "address_id" in values:
+            return {"partner_invoice_id": values["address_id"]}
+        else:
+            return {}
+
+    def _update_values(self, values):
+        if "shipping" in values:
+            shipping = values.pop("shipping")
+            if shipping:
+                values.update(self._convert_shipping_values(shipping))
+        if "invoicing" in values:
+            invoicing = values.pop("invoicing")
+            if invoicing:
+                values.update(self._convert_invoicing_values(invoicing))
+
+    def to_vals(self):
+        values = self.model_dump(exclude_unset=True)
+        self._update_values(values)
+        return values
```

Comment:
@sebastienbeau Take a look to the way it's done into the `sync` route. The logic to transform a model to an specific format to be compliant for a model method should go into a helper model. (for exemple: `shopinvader_api_cart.service.helper`)

Reply:
ok, I was hesitating if it's should be here or in the helper. I move it in the helper